### PR TITLE
feat(provider): add Mailchimp provider

### PR DESCRIPTION
**Pull request by michalbundyra**
**What**: add new provider - Mailchimp

**Why**: working in project where I am using this provider, so it might be useful also for others

**How**:

**Checklist**:

- [x] Documentation
- [ ] Tests N/A ?
- [x] Ready to be merged

### CODE REVIEW

File: src/providers/mailchimp.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+export default (options) => {
+  return {
+    id: 'mailchimp',
+    name: 'Mailchimp',
+    type: 'oauth',
+    version: '2.0',
+    scope: '',
```
Comment (balazsorban44) [inline src/providers/mailchimp.js:7]: Do you really not need to set any scopes here? 🤔

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Gotta love providers that implement OAuth in a way that "just works". 😁🎉 Nice PR, had a tiny comment, could you please check?

File: src/providers/mailchimp.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+export default (options) => {
```
Comment (balazsorban44) [inline src/providers/mailchimp.js:1]:
```suggestion
export default Mailchimp(options) {
```

**Comment by balazsorban44** [comment]
@michalbundyra my question stands about the scope. I wasn't curious if you can remove it but rather if it really works for you locally without providing any? I assumed that you would need something as basic as openid or something like that

**Comment by michalbundyra** [comment]
@balazsorban44 in my code I have empty string for scope and it works, but I'll verify it again and will be back here. Thanks

**Comment by balazsorban44** [comment]
thanks! could you please revert the src/providers/index.js file change? that file is auto-generated now

**Comment by michalbundyra** [comment]
@balazsorban44 I checked again documentation and there is no single word about the scope. It works for me with empty scope tho. 

It's now rebased, so merge conflicts are resolved. Thanks

**Review by balazsorban44 — APPROVED** [review]
Thank you so much!

**Pull request merged**